File: tests/src/main/java/com/abstratt/simon/examples/DAUI.java
package com.abstratt.simon.examples;

import com.abstratt.simon.examples.IM.Entity;
import com.abstratt.simon.metamodel.dsl.Meta;

@Meta.Package(builtIns = {})
public interface DAUI {
	interface IEntityComponent extends UI.IComponent {
		@Meta.Reference
		@Meta.Typed(Entity.class)
		public Entity entity();
	}
	
	class EntityScreen extends UI.Screen implements IEntityComponent {
		private Entity entity;

		@Override
		public Entity entity() {
			return entity;
		}
		
	}
}

File: example-languages-kotlin/src/main/kotlin/com/abstratt/simon/examples/kotlin/DAUI.kt
package com.abstratt.simon.examples.kotlin

import com.abstratt.simon.metamodel.dsl.Meta

@Meta.Package(builtIns = [])
interface DAUI {
    interface IEntityComponent : UI.IComponent {
        @Meta.Reference
        @Meta.Typed(IM.Entity::class)
        fun entity(): IM.Entity?
    }

    class EntityScreen : UI.Screen(), IEntityComponent {
        private val entity: IM.Entity? = null
        override fun entity(): IM.Entity {
            return entity!!
        }
    }
}